### PR TITLE
Docs: Add all types of files metadata tables in Spark queries

### PR DESCRIPTION
**Pull request by manuzhang**

### CODE REVIEW

File: docs/spark-queries.md [hunk 1]
```diff
@@ -270,17 +270,22 @@ order by made_current_at
...
-To show a table's current data files:
+To show a table's current files:
 
 ```sql
 SELECT * FROM prod.db.table.files;
 ```
 
-|content|file_path                                                                                                                                   |file_format|spec_id|partition|record_count|file_size_in_bytes|column_sizes      |value_counts    |null_value_counts|nan_value_counts|lower_bounds           |upper_bounds           |key_metadata|split_offsets|equality_ids|sort_order_id|
+| content | file_path | file_format | spec_id | partition | record_count | file_size_in_bytes | column_sizes | value_counts | null_value_counts | nan_value_counts | lower_bounds | upper_bounds | key_metadata | split_offsets | equality_ids | sort_order_id | readable_metrics |
 | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- |
```
Comment (dramaticlly) [inline docs/spark-queries.md:280]: alignment looks off, I think you want to match columns count here as well

image

Comment (manuzhang) [reply docs/spark-queries.md:280]: FIxed, please check again.

File: docs/spark-queries.md [hunk 2]
```diff
@@ -270,17 +270,22 @@ order by made_current_at
...
+| 1 | s3:/.../table/data/00081-4-a9aa8b24-20bc-4d56-93b0-6b7675782bb5-00001-deletes.parquet | PARQUET | 0 | 1 | 1560 | {2147483545:46,2147483546:152} | {2147483545:1,2147483546:1} | {2147483545:0,2147483546:0} | {} | {2147483545:,2147483546:s3:/.../table/data/00000-0-f9709213-22ca-4196-8733-5cb15d2afeb9-00001.parquet} | {2147483545:,2147483546:s3:/.../table/data/00000-0-f9709213-22ca-4196-8733-5cb15d2afeb9-00001.parquet} | NULL | [4] | NULL | NULL | {"data":{"column_size":null,"value_count":null,"null_value_count":null,"nan_value_count":null,"lower_bound":null,"upper_bound":null},"id":{"column_size":null,"value_count":null,"null_value_count":null,"nan_value_count":null,"lower_bound":null,"upper_bound":null}} |
+
+Content `0`  means it's a data file and content `1` a position-deletes file for [v2 format](../configuration/#reserved-table-properties). They also can be inspected individualy from `prod.db.table.data_files` and `prod.db.table.delete_files`.
```
Comment (aokolnychyi) [inline docs/spark-queries.md:286]: `content `1` a position-deletes file` -> `content `1` indicates a position delete file`. We also should document equality deletes.

File: docs/spark-queries.md [hunk 3]
```diff
@@ -270,17 +270,22 @@ order by made_current_at
...
+
+To show a table's files for all tracked snapshots, query `prod.db.table.all_files`, `prod.db.table.all_data_files` and `prod.db.table.all_delete_files` respectively.
```
Comment (aokolnychyi) [inline docs/spark-queries.md:288]: Question: Do we need `a` before `table's files`?

File: docs/spark-queries.md (hunk 2 above)
Comment (manuzhang) [reply docs/spark-queries.md:286]: 1. Are equality deletes typically used with Spark query? 
2. I'm copying `position-deletes file` from description in https://iceberg.apache.org/spec/#manifests
```
Type of content stored by the data file: data, equality deletes, or position deletes (all v1 files are data files)
```

File: docs/spark-queries.md (hunk 3 above)
Comment (manuzhang) [reply docs/spark-queries.md:288]: I'm going to change it to `files of a table`.

File: docs/spark-queries.md (hunk 2 above)
Comment (szehon-ho) [reply docs/spark-queries.md:286]: Even if not written, Spark query should be able to read them?  So I think makes sense to me to document that.

I think @aokolnychyi meant, to remove the hyphen from position-deletes-file (I dont see hyphen form in that doc)

Should we make a hint or note sub section for this block, and use bullet or numbers to be less repetitve?

```
Content refers to type of content stored by the data file:
  0  Data
  1  Position Deletes
  2  Equality Deletes
```

**Review by szehon-ho — COMMENTED** [review]
Thanks for updating the other files table and adding the readable_metrics

File: docs/spark-queries.md [hunk 4]
```diff
@@ -270,17 +270,30 @@ order by made_current_at
...
+| 2 | s3:/.../table/data/00047-25-833044d0-127b-415c-b874-038a4f978c29-00612.parquet | PARQUET | 0 | 126506 | 28613985 | {100:135377,101:11314} | {100:126506,101:126506} | {100:105434,101:11} | {} | {100:0,101:17} | {100:404455227527,101:23} | NULL | NULL | [1] | 0 | {"id":{"column_size":135377,"value_count":126506,"null_value_count":105434,"nan_value_count":null,"lower_bound":0,"upper_bound":404455227527},"data":{"column_size":11314,"value_count":126506,"null_value_count": 11,"nan_value_count":null,"lower_bound":17,"upper_bound":23}} |
+
+{{< hint info >}}
+Content refers to type of content stored by the data file:
+  0  Data
+  1  Position Deletes
+  2  Equality Deletes
+{{< /hint >}}
+
+They also can be inspected individualy from `prod.db.table.data_files` and `prod.db.table.delete_files`.
```
Comment (szehon-ho) [inline docs/spark-queries.md:294]: 'inspected individually' sounds like querying about one file, which is not the case.

How about:  `` To show only data files or delete files, query 'data_files' and 'delete_files' respectively.``

Then it matches the below sentence.

File: docs/spark-queries.md [hunk 5]
```diff
@@ -270,17 +270,30 @@ order by made_current_at
...
+
+To show files of a table for all tracked snapshots, query `prod.db.table.all_files`, `prod.db.table.all_data_files` and `prod.db.table.all_delete_files` respectively.
```
Comment (szehon-ho) [inline docs/spark-queries.md:296]: How about add a little more clarity, `` To show all files, data files, and delete files across all tracked snapshots, query all_files, all_data_files, and all_delete_files respectively.``

**Review by szehon-ho — APPROVED** [review]

**Comment by szehon-ho** [comment]
Merged, thanks @manuzhang

**Pull request merged**